contracts/account/paymaster/PaymasterCore.sol

```diff
@@ -0,0 +1,120 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {IEntryPoint, IPaymaster, PackedUserOperation} from "@openzeppelin/contracts/interfaces/draft-IERC4337.sol";
+import {ERC4337Utils} from "@openzeppelin/contracts/account/utils/draft-ERC4337Utils.sol";
+
+/**
+ * @dev A simple ERC4337 paymaster implementation. This base implementation only includes the minimal logic to validate
+ * and pay for user operations.
+ *
+ * Developers must implement the {PaymasterCore-_validatePaymasterUserOp} function to define the paymaster's validation
+ * and payment logic. The `context` parameter is used to pass data between the validation and execution phases.
+ *
+ * The paymaster includes support to call the {IEntryPointStake} interface to manage the paymaster's deposits and stakes
+ * through the internal functions {_deposit}, {_withdraw}, {_addStake}, {_unlockStake} and {_withdrawStake}.
+ *
+ * * Deposits are used to pay for user operations.
+ * * Stakes are used to guarantee the paymaster's reputation and obtain more flexibility in accessing storage.
+ *
+ * NOTE: See [Paymaster's unstaked reputation rules](https://eips.ethereum.org/EIPS/eip-7562#unstaked-paymasters-reputation-rules)
+ * for more details on the paymaster's storage access limitations.
+ */
+abstract contract PaymasterCore is IPaymaster {
+    /// @dev Unauthorized call to the account.
```

Comment:
IMO this error (and the equivalent in AccountCore) should be renamed to something like `EntryPointRestricted(address caller)`

```suggestion
    /// @dev Unauthorized call to the paymaster.
```